site/app/controllers/admin/ReportController.php

```diff
@@ -259,7 +259,7 @@ private function mergeGradedGradeables(array $gradeables, User $user, array $use
                 $graded_gradeable = $user_graded_gradeables[$g->getId()];
             }
 
-            $graded_gradeable->setOverriddenGrades($this->all_overrides[$user->getId()][$graded_gradeable->getGradeableId()] ?? null);
+            $graded_gradeable?->setOverriddenGrades($this->all_overrides[$user->getId()][$graded_gradeable->getGradeableId()] ?? null);
             $ggs[] = $graded_gradeable;
```

Comment:
You probably want to wrap this in a null check instead of the null access operator because you're appending a null gradeable to the list.

```suggestion
            if($graded_gradeable != null) {
                $graded_gradeable->setOverriddenGrades($this->all_overrides[$user->getId()][$graded_gradeable->getGradeableId()] ?? null);
                $ggs[] = $graded_gradeable;
            }
```

Reply:
You are correct, I initially thought of wrapping with if statement, then while trying to optimize the code, missed about appending to ggs. Thanks for the catch, I will commit your suggestion, and change the if condition to more strict comparison.